gh-ci-pipeline/src/main/java/Main.java

```diff
@@ -0,0 +1,190 @@
+        return creds.getCredentials();
+    }
+
+    // Bad
+    public static void run1() {
+        try {
+            Cipher cipher = Cipher.getInstance(CIPHER);
+            System.out.println(cipher);
+        }
+        catch (NoSuchPaddingException e) {
+            e.printStackTrace();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Good
+    public static void run2() {
+        try {
+            Cipher cipher = Cipher.getInstance(CIPHER_GOOD);
+            System.out.println(cipher);
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        } catch (NoSuchPaddingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    //Bad
+    public static void createCookie1(final HttpServletResponse response) {
+        Cookie cookie = new Cookie("name", "value");
+        response.addCookie(cookie);
+
+    }
+
+    // Good
+    public static void createCookie2(final HttpServletResponse response) {
+        Cookie cookie = new Cookie("name", "value");
+        cookie.setSecure(true);
+        response.addCookie(cookie);
+    }
+
+    // Bad
+    public static void reEncrypt1() {
+        AWSKMS client = AWSKMSClientBuilder.standard().build();
+        ByteBuffer sourceCipherTextBlob = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0});
+
+        DecryptRequest req = new DecryptRequest()
+            .withCiphertextBlob(sourceCipherTextBlob);
+        ByteBuffer plainText = client.decrypt(req).getPlaintext();
+
+        EncryptRequest res = new EncryptRequest()
+            .withKeyId("NewKeyId")
+            .withPlaintext(plainText);
+        ByteBuffer ciphertext = client.encrypt(res).getCiphertextBlob();
+        System.out.println(ciphertext);
+    }
+
+    // Good
+    public static void reEncrypt2() {
+        ByteBuffer sourceCipherTextBlob = ByteBuffer.wrap(new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0});
+
+        ReEncryptRequest req = new ReEncryptRequest()
+            .withCiphertextBlob(sourceCipherTextBlob)
+            .withDestinationKeyId("NewKeyId");
+        ByteBuffer ciphertext = req.getCiphertextBlob();
+        System.out.println(ciphertext);
+    }
+
+    private String decode(final String val, final String enc) {
+        try {
+            return java.net.URLDecoder.decode(val, enc);
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+        return "";
+    }
+
+    // Bad
+    public void pathTraversal1(HttpServletRequest request) throws IOException {
+        javax.servlet.http.Cookie[] theCookies = request.getCookies();
+        String path = "";
+        if (theCookies != null) {
+            for (javax.servlet.http.Cookie theCookie : theCookies) {
+                if (theCookie.getName().equals("thePath")) {
+                    path = decode(theCookie.getValue(), "UTF-8");
+                    break;
+                }
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 139:146, 161:168.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.